src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java

```diff
@@ -36,13 +35,17 @@ public UserRetrievalController(UserService userService) {
      *
      * @param userId
      * @return
-     * @throws IdNotFoundException
+     * @throws NotFoundException
      */
     @GetMapping("/user/{userId}")
     public ResponseEntity<UserResponse> retrieveUser(@PathVariable("userId") @Valid String userId) {
         User user = userService.getUserByUserId(userId);
-        UserResponse userResponse = UserResponse.builder().success(true).id(user.getId()).userName(user.getUserName())
-                .firstName(user.getFirstName()).lastName(user.getLastName()).build();
+        UserResponse userResponse = UserResponse.builder().success(true).id(user.getId()).userName(user.getUserName()).firstName(user.getFirstName()).lastName(user.getLastName()).build();
         return new ResponseEntity<>(userResponse, HttpStatus.OK);
     }
+
+    @GetMapping("/User/{userName}")
+    public ResponseEntity<AppResponse> retrieveUserByName(@PathVariable("userName") @Valid String userName) {
+        return new ResponseEntity<>(userService.getUserByName(userName), HttpStatus.OK);
```

Comment:
Follow similar pattern for methods in the controller, here the service class is returning the response object, however in the above method, the controller class is building the response object. Please make them similar.

Reply:
As the above method is done by Vinod  and as he got comment from Shiju to return response on that pattern so now all the methods have changed to similar pattern